components/create_post/index.js

```diff
@@ -112,6 +116,8 @@ function makeMapStateToProps() {
             canPost,
             useChannelMentions,
             shouldShowPreview: showPreviewOnCreatePost(state),
+            allowReferencedGroups: getAllAssociatedGroupsForReference(state),
```

Comment:
thoughts on converting this into a map of the format 
`{groupAName: groupA, groupBName: groupB}` to make some of the computations faster when calling groups mentioned in text?

Reply:
Prodigy thoughts Farhan! Thanks for this! I implemented using map instead.